### DISCUSSION TITLE
Anschluss Pylontech CAN

### DISCUSSION THREAD

**Discussion by dome721** [hoylabs/OpenDTU-OnBattery#1000] [Q&A] [answered]
Hallo Community, 
ich hab schon im Wiki und den Issues/Discussions gesucht, habe aber keine Antwort gefunden (falls das Thema schon geschrieben wurde bitte verlinken).
Ich habe das System schon länger am laufen und möchte dies nun um einen Speicher ergänzen.
Dabei habe ich einen 48V Akku mit 100Ah, verbaut ist ein PACE BMS, das CAN Protokoll ist auf Pylontech konfiguriert. 
Meine Frage ist zur PIN Belegung des SN65HVD230 am ESP. Kann ich die Pins frei wählen, oder müssen die Pins spezielle Eigenschaften haben?
In der Konsole wird kein Kommunikationsfehler angezeigt, gibt hier es einen Log, wenn die Kommunikation nicht erfolgreich ist?

Vielen Dank schonmal für eure Antworten.

**Comment by Manos1966** [comment, marked as answer]
Hallo @dome721 
Du hast Recht, die Antwort ist gut verstackt 😉 

> Kann ich die Pins frei wählen, oder müssen die Pins spezielle Eigenschaften haben?

https://github.com/helgeerbe/OpenDTU-OnBattery/wiki/Device-Profiles#input-only--output-only-pins
wir werden den Link im Q&A addierern...

Der Rest ist hier:
https://github.com/helgeerbe/OpenDTU-OnBattery/wiki/Pylontech-battery#can-bus-connection

> In der Konsole wird kein Kommunikationsfehler angezeigt, gibt hier es einen Log, wenn die Kommunikation nicht erfolgreich ist?

"Ausfuehrliche Protokollierung" einschalten

**Reply by dome721** [reply]
Danke für deine Antwort, letztendlich war es nicht der ESP, bzw. das Pinmapping.
Da ich immer wieder Verbindungsprobleme zum HMS-1800 hatte, habe ich mir das DTU Fusion Board mit Erweiterung für Ve.Direct und CAN gekauft. 
Leider war in der Dokumentation meines Speichers eine falsche Pinbelegung zudem gab es über die BMS Software am Laptop ein anderes Pylontech CAN Protokoll (mit Index V202), mittlerweile klappt sowohl die Kommunikation zum Victron als auch zum Speicher.

**Reply by moserbe** [reply]
ich habe das Gefühl, ich könnte an einem ähnlichen Problem sein. Hast du die Pinbelegung noch und kannst sie veröffentlichen? Ich hab noch einen neuen Speicher und will den (noch) nicht öffnen  um zu schauen, wo die Drähte dran sind.

**Reply by dome721** [reply]
![image](https://github.com/user-attachments/assets/98294be4-415b-4bc2-98de-8c33bfe68ed5)

So steht es in meiner Bedienungsanleitung des Speichers, CAN H und CAN L waren bei mir vertauscht.
In meinem Speicher ist ein Pace BMS P16S100A verbaut.

So ist bei mir die richtige Pin Belegung sein:
![2 png 7469a881576b13fbc1277ebb6d9c97d6](https://github.com/user-attachments/assets/92c65073-6949-45e1-a322-789f03ce3a2e)

Ich hoffe, dass hilft Dir weiter! Hast Du die Software um das BMS zu konfigurieren?

**Reply by moserbe** [reply]
Danke erstmal! Ja, habe auch ein  Pace BMS P16S100A. 

Leider ist bei mir die Doku vermutlich dann nicht falsch. Meine Doku sagt 4: CAN H, 5: CAN L (und das scheint ja auch der überwiegende Standard zu sein). 

Ja, habe eine BMS Software zum Konfigurieren - habe auch schon deinen Vorschlag probiert, das PYLON_V202 zu verwenden. Hat nix gebracht. Ich probiere jetzt mal das Fusion Board (hab ich mir geshoppt). Vielleicht klappt es ja damit und ich habe ein defektes oder falsch verdrahtetes CAN Modul.

**Reply by moserbe** [reply]
So, ich habe mein Problem gefunden. War keine falsche Beschriftung, ware keine falsche Verdrahtung, keine defekten Bauteile.

Ich verwende ein PACE BMS mit dem Pylontech Protokoll über CAN. .... und das PACE scheine einen Bug oder Feature zu haben. 

Lt. https://www.akkudoktor.net/forum/stell-dein-batterie-powerwall-projekt-vor/gobel-power-51-2v-280ah-15kwh-lifepo4-server-rack-battery/paged/6/#post-91142 und https://diysolarforum.com/threads/help-with-pace-bms-comms.70940/ meldet das BMW über CAN NICHTS, wenn die DIP Schalter auf 0 stehen. 

Also den ersten DIP hoch schieben...und schon klappt es. ;-) 

Hat mich Tage gekostet, hoffe ihr erspart auch das!
